src/runtime_safe_heap.js

```diff
@@ -25,9 +25,7 @@ var SAFE_HEAP_COUNTER = 0;
 
 /** @param {number|boolean=} isFloat */
 function SAFE_HEAP_STORE(dest, value, bytes, isFloat) {
-#if CAN_ADDRESS_2GB
   dest >>>= 0;
-#endif
```

Comment:
This will not work for wasm64, I think?

It is harmless for wasm32 with less than 2GB maximum address though, and code size is not an issue for this type of build. But I also don't see how it would help? If we can't address over 2GB then we don't need this, I think?

Reply:
Ah yes, good point.

The specific issue it about read the very last word of the 2gb region.  Apparently this result is the check failing because the end address overflows.  (see the linked bug).